src/com/ConnectionInfoPublisher.cpp

```diff
@@ -0,0 +1,93 @@
+#include "ConnectionInfoPublisher.hpp"
+#include <chrono>
+#include <thread>
+#include <boost/uuid/name_generator.hpp>
+#include <boost/uuid/string_generator.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/filesystem.hpp>
+
+ConnectionInfoPublisher::ConnectionInfoPublisher(std::string acceptorName,
+                                                 std::string requesterName,
+                                                 int rank,
+                                                 std::string addressDirectory)
+  :
+  acceptorName(acceptorName),
+  requesterName(requesterName),
+  rank(rank),
+  addressDirectory(addressDirectory)
+{}
+  
+
+ConnectionInfoPublisher::ConnectionInfoPublisher(std::string acceptorName,
+                                                 std::string requesterName,
+                                                 std::string addressDirectory)
+  :
+  acceptorName(acceptorName),
+  requesterName(requesterName),
+  addressDirectory(addressDirectory)
+{}
+
+
+std::string ConnectionInfoPublisher::getFilename()
```

Comment:
I have a few comments here:

1) Doesn't this function return a relative path to the file? (misleading name)
2) Is there a reason why this function doesn't return a `boost::filesystem::path` instead of a `string`?
3) All dependencies of this function are const members, so this can be calculated in the constructor.
4) This is operation is semantically `const`
```suggestion
std::string ConnectionInfoPublisher::getFilename() const
```

Reply:
> All dependencies of this function are const members, so this can be calculated in the constructor.

It's called in both overloads of the constructor, so this will result in code duplication.

Reply:
Sorry, I don't understand what you mean. In the current state, it is called in the destructor as well as the read, and write.
It also doesn't modify the object, so there is no reason not make it `const`.

What I actually meant with point 3 is the following:

By design, this class cannot change its members after construction.
This function depends only on the members of the class.
Thus, once constructed, this member function will always return the same value.
Meaning, you can simply generate this value one in the ctor and use it later.